lib/system/fs.go

```diff
@@ -54,7 +54,7 @@ func GetFilesystem(ctx context.Context, path string, runner utils.CommandRunner)
 // GetFilesystemForPath returns the filesystem type for given path.
 // It does not verify whether the path actually exists
 func GetFilesystemForPath(path string) (fstype string, err error) {
-	mounts, err := mount.GetMounts()
+	mounts, err := mount.GetMounts(mount.ParentsFilter(path))
```

Comment:
Does this render the code below (or parts of it) redundant?

Reply:
ParentsFilter returns all mountpoints which can be parents for given path. It decreases amount of returned mounts to a few, but we still need logic above to traverse path till we find real mountpoint.